### PR TITLE
fix: duplicated messages logged at INFO and ERROR levels.

### PR DESCRIPTION
**Pull request by CanmingCobble**
## Description

This change will fix logging the same exit error message at the INFO level as well as at the ERROR level. In addition, this fix cherry-picked a change made for FOUNDENG-260 from https://github.com/determined-ai/determined-ee/pull/463.

## Test Plan
With this change:

 % det command run exit 0
[2022-10-24T16:12:58.439433Z]          || INFO: Scheduling Command (needlessly-included-thrush) (id: 9765c6e7-d221-464f-81da-f8b6e3c17302)

[2022-10-24T16:12:58.580698Z]          || INFO: Command (needlessly-included-thrush) was assigned to an agent
[2022-10-24T16:13:19.466281Z]          || INFO: resources exited successfully with a zero exit code
[2022-10-24T16:13:19.508887Z]          || INFO: Command (needlessly-included-thrush) was terminated: allocation stopped early after resources exited successfully with a zero exit code

% det command run exit 1
[2022-10-24T16:13:28.448415Z]          || INFO: Scheduling Command (actively-amused-drum) (id: a7718806-3125-42d7-8e94-94100ea68688)
[2022-10-24T16:13:28.581021Z]          || INFO: Command (actively-amused-drum) was assigned to an agent
[2022-10-24T16:13:48.396415Z]          || ERROR: Command (actively-amused-drum) was terminated: allocation failed: task failed without an associated exit code: Slurm job is in a failed state due to reason 'NonZeroExitCode':
  Job Exited with Exit Code 1
  INFO:    Using cached SIF image
  INFO: Setting workdir to /run/determined/workdir
  INFO: executing /run/determined/command-entrypoint.sh exit 1
  srun: error: node002: task 0: Exited with exit code 1
  srun: launch/slurm: _step_signal: Terminating StepId=1492.0. SBATCH options: --ntasks-per-node=1 --nodes=1-1 --gpus=1 --job-name=det-ai_cmd-9dfed72b568d4ffe-9229be15f4d022f9 --partition=defq --chdir=/var/tmp --error=/home/users/cobble/.launcher/jobs/environments/cobble/9dfed72b568d4ffe-9229be15f4d022f9/ai_cmd-error.log --output=/home/users/cobble/.launcher/jobs/environments/cobble/9dfed72b568d4ffe-9229be15f4d022f9/ai_cmd-output.log
 (exit code 1)

image

## Checklist
- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: master/internal/task/allocation.go [hunk 1]
```diff
@@ -499,7 +499,7 @@ func (a *Allocation) SetResourcesAsDaemon(
 		a.logger.Insert(ctx, a.enrichLog(model.TaskLog{
 			Log: `Ignoring request to daemonize resources within an allocation for an allocation
 			with only one manageable set of resources, because this would just kill it. This is
-			expected in when using Slurm.`,
+			expected in when using the HPC launcher.`,
```
Comment (CanmingCobble) [inline master/internal/task/allocation.go:502]: Change cherry-picked from PR https://github.com/determined-ai/determined-ee/pull/463

**Review by jerryharrow — APPROVED** [review]
lgtm -- let's see what @stoksc  thinks.

File: master/internal/sproto/task.go [hunk 2]
```diff
@@ -272,8 +273,12 @@ func (ev *Event) ToTaskLog() model.TaskLog {
 		message = ""
 	}
 
+	level := ptrs.Ptr(model.LogLevelInfo)
+	if ev.Level != nil {
```
Comment (rcorujo) [inline master/internal/sproto/task.go:277]: What is the meaning of "ev.Level" and why do we overwrite "level" with "ev.Level"?

Comment (CanmingCobble) [reply master/internal/sproto/task.go:277]: Event message severity level. All event messages were logged at LogLevelInfo, including the failure message this PR is fixing.

Comment (rcorujo) [reply master/internal/sproto/task.go:277]: What if "ev.Level" is TRACE or DEBUG, assuming those are valid values?  Are we still OK?

Comment (CanmingCobble) [reply master/internal/sproto/task.go:277]: let's see what @stoksc thinks.

Comment (stoksc) [reply master/internal/sproto/task.go:277]: if they are trace or debug, it seems the sender could send them at that level now? and info is a fine default. i think this is ok.

**Review by stoksc — APPROVED** [review]
lgtm

**Pull request merged**